### ISSUE TITLE
Create a virtual_network_gateway with azurerm

### ISSUE THREAD

**Issue by hashibot** [terraform-providers/terraform-provider-azurerm#17]
_This issue was originally opened by @dpnl87 as hashicorp/terraform#8372. It was migrated here as part of the [provider split](https://www.hashicorp.com/blog/upcoming-provider-changes-in-terraform-0-10/). The original body of the issue is below._

<hr>

I would like to have to possibility to create virtual network gateways with Terraform and the Azurerm provider. Currently there is no support for it so we still need to do manual actions on Azure.

Its used to link a network interface to a ExpressRoute.

If you need any additional information or require testing assistance I'm more then happy to help out.

**Comment by jcwest** [comment]
@dominik-lekse Thank you for working on this. My team needs these resources.
It seems that there hasn't been much movement in almost a month. Please let me know if there is any way we can help make this feature available sooner.

**Comment by samirageb** [comment]
Also just ran into this missing capability in TF. What makes it unfortunate apparently VnetGW are required to exist in the same resource group as the VNet they're connected to...  🙁  So managing vnets/subnets with TF is a forced config drift situation...  (Wishing I knew go and could contrib right about now...)

**Comment by tombuildsstuff** [comment]
Hey @samirageb 

> Also just ran into this missing capability in TF

This has been implemented by @dominik-lekse in PR #133 - we're just waiting on Microsoft to fix an API before we can proceed (which is scheduled for mid-month)

> What makes it unfortunate apparently VnetGW are required to exist in the same resource group as the VNet they're connected to

That's good to know - do we add something to that effect in the documentation @dominik-lekse?

> So managing vnets/subnets with TF is a forced config drift situation.

That shouldn't be an issue - would it be possible to open a separate issue to discuss that?

Thanks!

**Comment by samirageb** [comment]
@tombuildsstuff Thanks for the updates.

Re: config drift, I was simply referring to the fact that VNETs/Subnets can be controlled via TF, but VNetGW's can't, so I have RG's that are partially managed by TF. (No new issue req'd)

**Comment by mud5150** [comment]
@samirageb I would also really like this to be finished and merged however there is the work-around of using the arm template resource. I'm in the process of getting this working myself. If it would be helpful I could post a gist of an example configuration.

**Comment by kmcquade** [comment]
@mud5150 sample repo for that on my page. VPC peering not included but that’s a simple addition. Won’t be useful after they release this fix but might help you out in the meantime

**Comment by kmcquade** [comment]
I suggest closing this issue. Duplicate and it has now been addressed as of release v1.1.0

@tombuildsstuff just because I know you have privs to close :)

**Comment by tombuildsstuff** [comment]
Thanks for the ping @kmcquade :)
